src/hotspot/share/c1/c1_Optimizer.cpp

```diff
@@ -1198,6 +1201,28 @@ void NullCheckEliminator::handle_ProfileReturnType(ProfileReturnType* x) {
   x->set_needs_null_check(!set_contains(x->ret()));
 }
 
+void NullCheckEliminator::handle_Constant(Constant *x) {
+  ObjectType* ot = x->type()->as_ObjectType();
+  if (ot && ot->is_loaded()) {
```

Comment:
Hotspot style guide insists we avoid implicit bool conversions. Check `ot != nullptr` explicitly.

Reply:
fixed.